monai/transforms/utils.py

```diff
@@ -732,6 +734,65 @@ def get_largest_connected_component_mask(img: torch.Tensor, connectivity: Option
     return torch.as_tensor(largest_cc, device=img.device)
 
 
+def fill_holes(
+    img_arr: np.ndarray, applied_labels: Optional[Iterable[int]] = None, connectivity: Optional[int] = None
+) -> np.ndarray:
+    """
+    Fill the holes in the provided image.
+
+    The label 0 will be treated as background and the enclosed holes will be set to the neighboring class label.
+    What is considered to be an enclosed hole is defined by the connectivity.
+    Holes on the edge are always considered to be open (not enclosed).
+
+    Note:
+
+        The performance of this method heavily depends on the number of labels.
+        It is a bit faster if the list of `applied_labels` is provided.
+        Limiting the number of `applied_labels` results in a big decrease in processing time.
+
+        If the image is one-hot-encoded, then the `applied_labels` need to match the channel index.
+
+    Args:
+        img_arr: numpy array of shape [C, spatial_dim1[, spatial_dim2, ...]].
+        applied_labels: Labels for which to fill holes. Defaults to None,
+            that is filling holes for all labels.
+        connectivity: Maximum number of orthogonal hops to
+            consider a pixel/voxel as a neighbor. Accepted values are ranging from  1 to input.ndim.
+            Defaults to a full connectivity of ``input.ndim``.
+
+    Returns:
+        numpy array of shape [C, spatial_dim1[, spatial_dim2, ...]].
+    """
+    channel_axis = 0
+    num_channels = img_arr.shape[channel_axis]
+    is_one_hot = num_channels > 1
+    spatial_dims = img_arr.ndim - 1
+    structure = ndimage.generate_binary_structure(spatial_dims, connectivity or spatial_dims)
+
+    # Get labels if not provided. Exclude background label.
+    applied_labels = set(applied_labels or (range(num_channels) if is_one_hot else np.unique(img_arr)))
+    background_label = 0
+    applied_labels.discard(background_label)
+
+    for label in applied_labels:
+        tmp = np.zeros(img_arr.shape[1:], dtype=bool)
+        ndimage.binary_dilation(
+            tmp,
+            structure=structure,
+            iterations=-1,
+            mask=np.logical_not(img_arr[label]) if is_one_hot else img_arr[0] != label,
+            origin=0,
+            border_value=1,
+            output=tmp,
+        )
+        if is_one_hot:
+            img_arr[label] = np.logical_not(tmp)
```

Comment:
Hi @wyli @rijobro , I think for the One-Hot case, we should suppose `applied_labels` as channel indices instead of pure label value, because One-Hot only has values (0, 1), what do you think?

Thanks.

Reply:
@Nic-Ma This is how it is implemented, or are you asking if this is acceptable?

Reply:
yes, looks good to me